app/core/pokemon-entity.ts

```diff
@@ -1762,16 +1763,26 @@ export class PokemonEntity extends Schema implements IPokemonEntity {
     this.life = this.refToBoardPokemon.hp
     this.shield = 0
     this.pp = 0
-    this.ap = 0
+    this.ap = this.refToBoardPokemon.ap
     this.atk = this.refToBoardPokemon.atk
     this.def = this.refToBoardPokemon.def
     this.speDef = this.refToBoardPokemon.speDef
     this.atkSpeed = this.refToBoardPokemon.atkSpeed
     this.critChance = DEFAULT_CRIT_CHANCE
     this.critPower = DEFAULT_CRIT_POWER
+    this.atkSpeed = DEFAULT_ATK_SPEED
+    this.range = this.refToBoardPokemon.range
+    this.dodge = 0
+    this.luck = 0
     this.count = new Count()
-    this.status.clearNegativeStatus()
     this.effects.clear()
+    this.status = new Status({
```

Comment:
should probably be a clear() method instead of losing the existing instance. Why not `clearNegativeStatus` ?

Reply:
Status isn't a set so it can't be directly cleared. `clearNegativeStatus` doesn't encompass all of the effects that need to be cleared. There are far fewer properties that need to be retained than those that need to be reset, so it's easier to retain those fewer properties.

Reply:
I'm not referring to a Set, i'm referring to declaring a method to reset the statuses properly on resurrect. I also believe its for the best to manually specify which statuses should be reset on resurrection, because we may want to preserve some. For example, the enraged status should not be reset after 35 seconds of fight. Other passive-linked statuses like Clear Wing should not be cleared neither.

Reply:
How is Clear Wing any different from Ground stacks or Soul Dew? It's not obvious why passives shouldn't be reset. 

I thought that enraged was fine to be reset because it would just be triggered again, but maybe with update delay it's not? I think it's fine to specify what to reset instead of specifying what to keep, but here's my reasoning for why I decided to specify what to keep:
The vast majority of statuses interact with only the pokemon they are attached to, so they can be reset on death. However, there are a few statuses which, as a special case, interact with the board state in some way. Ghost curses need to transmit to other pokemon and enraged needs to trigger based on the round timer. Because these cases are the exception and not the norm, it makes more sense that they are the ones explicitly stated.